main.js

```diff
@@ -13,6 +13,357 @@ toggle.addEventListener('click', () => {
+txt.className = popPara;
+poptext.appendChild(txt);
+
+const bottomBtnDiv = createDiv(popupData[0].divCls[4]);
+const btn3 = createBtn(popupData[0].btns[2]);
+btn3.className = seeLive;
+btn3.innerHTML = demo;
+bottomBtnDiv.appendChild(btn3);
+
+const btn4 = createBtn(popupData[0].btns[3]);
+btn4.className = seeLink;
+btn4.innerHTML = src;
+bottomBtnDiv.appendChild(btn4);
+poptext.appendChild(bottomBtnDiv);
+popDiv.appendChild(poptext);
+modal.appendChild(popDiv);
+
+const cardBtn = document.querySelectorAll('.cardBtn');
+const body = document.querySelector('#portfolio');
+const hero = document.querySelector('.hero');
+const popup = document.querySelector('.popup');
+const collapse = document.querySelector('.fa-x');
+
+cardBtn.forEach((item) => {
+  item.addEventListener('click', () => {
+    popup.classList.add('open');
+    body.classList.add('blur');
+    hero.classList.add('blur');
+  });
+});
+
+collapse.addEventListener('click', () => {
+  popup.classList.remove('open');
+  body.classList.remove('blur');
+  hero.classList.remove('blur');
+});
+
+const demoLink = document.querySelectorAll('.seeLive');
+const source = document.querySelectorAll('.seeLink');
+
+demoLink.forEach((item) => {
+  item.addEventListener('click', () => {
+    window.location.href = gitpages;
+  });
+});
+
+source.forEach((item) => {
+  item.addEventListener('click', () => {
+    window.location.href = github;
+  });
+});
+
+const nameErr = document.querySelector('.fullNameError');
+const emailErr = document.querySelector('.emailError');
+const fullNValid = document.querySelector('.fullNValid');
+const emailValid = document.querySelector('.emailValid');
+const msgErr = document.querySelector('.msgError');
+const msgValid = document.querySelector('.msgValid');
+const submitErr = document.querySelector('.submitErr');
+
+function validateFullName() {
+  const fullName = document.getElementById('fullname').value;
+
+  if (fullName.length === 0) {
+    nameErr.innerHTML = 'Full name is required';
+    fullNValid.innerHTML = null;
+    return false;
+  }
+  if (!fullName.match(/^[A-Za-z]*\s{1}[A-Za-z]*$/)) {
+    nameErr.innerHTML = 'Enter second name';
+    fullNValid.innerHTML = null;
+    return false;
+  }
+  nameErr.innerHTML = null;
+  fullNValid.innerHTML = '<i class="fa-solid fa-circle-check"></i>';
+  return true;
+}
+
+function validateEmail() {
+  const email = document.getElementById('email').value;
+
+  if (email.length === 0) {
+    emailErr.innerHTML = 'Email is required';
+    emailValid.innerHTML = null;
+    return false;
+  }
+  if (
+    !email.match(
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+    )
+  ) {
+    emailErr.innerHTML = 'Email Invalid';
+    emailValid.innerHTML = null;
+    return false;
+  }
+  emailErr.innerHTML = null;
+  emailValid.innerHTML = '<i class="fa-sharp fa-solid fa-circle-check"></i>';
+  return true;
+}
```

Comment:
- [ Required ] Please consider adding a validation that checks if the email address provided by a user is in `lowercase` or `uppercase` as stated in the project requirements. Currently, I'm able to submit the contact form when the `email` field is in `uppercase` which should not be the case, please see the screenshot below and rectify this.

## Project requirements
![Screenshot from 2023-02-10 12-05-13](https://user-images.githubusercontent.com/16392046/218050752-60719800-abbe-4dae-9da5-e35afedafbf8.png)